### DISCUSSION TITLE
Dax Studio vs Power Query

### DISCUSSION THREAD

**Discussion by Machestro** [DaxStudio/DaxStudio#1202] [General]
Why DAX Studio extracts data from tabular SSAS model much faster than Power Query with the same DAX code? Perhaps you know the reason, and. Is therer as way to speed up PQ?. Perhaps the reason is in different connection drivers (OLEDEB, ADO...)

**Comment by marcosqlbi** [comment]
Because - in general - Power Query is slower compared to native code.
The first version of [VertiPaq Analyzer](https://www.sqlbi.com/tools/vertipaq-analyzer/) was an Excel file extracting data using Power Query transformations. Because with large models it was slow, and because of the limitations in Power Query transformations, I wrote the second version as a [library published on GitHub](https://github.com/sql-bi/VertiPaq-Analyzer) that can be used in many tools to extract the information with much more details thanks to the more complex transformation and compatibility changes that the library can manage. DAX Studio is one of the several tools that take advantage of those libraries.
